### ISSUE TITLE
Need an error and event log

### ISSUE THREAD

**Issue by oyeb** [arrow-/phoenix#22]
### Currently
bot's STDERR -> debug_log
bot's STDOUT -> move_log

### What an error_log might look like
What if the bot is making an invalid move? We register an exception on the `gameloop` STDOUT but the bot is never told that. The gameloop can log _errors-that-happen-due-to-bots_ in a error_log.
There could be a error_log per bot.
**This log cannot be seen by the bot, in any iteration. Should we send the _errors-in-prev-iter_ to the bot as part of the new game-state?**

### What an event log might look like
There is some event logging in `gameloop` STDOUT, a tuple of (turn_no, [list-of-events]) would be a compact event log (Since `turn_no` is explicit, we don't need a row for empty turns).
**This log cannot be seen by the bot, in any iteration. Should we send the _errors-in-prev-iter_ to the bot as part of the new game-state?**

**Comment by oyeb** [comment]
[error_log]
`Gamectl.is_valid_move()` can be made to return errors to `gameloop`, which can log it in the correct file

**Comment by vasanthaganeshk** [comment]
+ We just kill the bot, that makes an invalid move (its still on map though!).
+ Creating event log is necessary (high priority, probably)
+ logging errors, that can be useful for debugging is a total mess. Returning errors to gameloop and logging it appropriately can be a start to fix this issue.

**Comment by oyeb** [comment]
Yes, I noticed `bot.vaild = False`. IMO it's a very high penalty for a single invalid move.
All we have to do is return the traceback objects, or certain helpful string messages, we are catching these errors anyways.
